rewriter/SourceRewriter.cpp

```diff
@@ -798,6 +846,52 @@ int main(int argc, const char **argv) {
                        options_parser.getSourcePathList());
   CompilationDatabase &comp_db = options_parser.getCompilations();
 
+  for (auto s : options_parser.getSourcePathList()) {
+    assert(s.starts_with(RootDirectory));
+    std::string relative_path = s.substr(RootDirectory.length());
+    std::string new_path = OutputDirectory;
+    new_path.append(relative_path);
+
+    std::string parent_dir = new_path;
+    auto n = parent_dir.rfind('/');
+    parent_dir = parent_dir.substr(0, n);
+    if (!opendir(parent_dir.c_str())) {
+      auto cmd = "mkdir -p "s + parent_dir;
+      system(cmd.c_str());
```

Comment:
`system()` always scares me because we're passing an input param to a shell. Would it be too hard to recursively check parents and call `mkdir(3)`?

Reply:
No, that should be fine and was what I initially started doing but I just wanted to get something that worked first. I can switch it out.